src/mod.ts

```diff
@@ -91,7 +99,7 @@ export async function createPlugin(
     );
   }
 
-  const [names, moduleData] = await _toWasmModuleData(await Promise.resolve(manifest), opts.fetch ?? fetch);
+  const [names, moduleData] = await _toWasmModuleData(m, opts.fetch ?? fetch);
```

Comment:
In order to support manifest options when the manifest is provided via `fetch`, we might rework `_toWasmModuleData` to return a third item, `manifestOpts`, then perform this call before normalizing the options in this function.

[`intoManifest`](https://github.com/extism/js-sdk/blob/main/src/manifest.ts#L79) is the chokepoint at which all the disparate input types are unified into something "manifest-like", so we could extract manifest-provided options [here](https://github.com/extism/js-sdk/blob/main/src/manifest.ts#L160) and then use them in `createPlugin` to finish normalizing the input options.

Reply:
is this what you meant?

https://github.com/extism/js-sdk/blob/d3e0dbcb9575f8c93307bca0f03a6e31cf3ab132/src/manifest.ts#L97

Reply:
Yep! That's perfect, thank you!